src/v/cluster/partition.h

```diff
@@ -281,11 +315,19 @@ class partition {
     ss::sharded<features::feature_table>& _feature_table;
     bool _is_tx_enabled{false};
     bool _is_idempotence_enabled{false};
+    ss::lw_shared_ptr<archival::configuration> _archival_conf;
+    ss::sharded<cloud_storage::remote>& _cloud_storage_api;
+
     ss::shared_ptr<cloud_storage::remote_partition> _cloud_storage_partition;
+    ss::lw_shared_ptr<archival::ntp_archiver> _archiver;
```

Comment:
I'd really like it if this could be a `unique_ptr`.  `partition` now owns the archiver and it would be nice to reflect this in the types.

Reply:
Yep, quite right.  I've changed this.